code/datums/uplink_items/uplink_traitor.dm

```diff
@@ -574,10 +576,11 @@
 	var/cost
 
 /datum/uplink_item/bundles_TC/surplus_crate/spawn_item(turf/loc, obj/item/uplink/U)
+	if(..() != UPLINK_SPECIAL_SPAWNING)
+		return FALSE
+
 	var/obj/structure/closet/crate/surplus/surplus = new(loc, U, crate_value)
 	surplus.cost = cost
-	U.uses -= cost
-	U.used_TC = cost
 
 /obj/structure/closet/crate/surplus/Initialize(mapload, obj/item/uplink/U, crate_value)
```

Comment:
Can you move all of the object stuff to another file? Perhaps crates.dm. Just keep the spawning stuff here.